integration-tests/tests/src/tests/sets.ts

```diff
@@ -0,0 +1,493 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import Realm from "realm";
+import { openRealmBeforeEach } from "../hooks";
+
+describe("Sets", () => {
+  beforeEach(Realm.clearTestState);
+
+  const IntSetObjectSchema = {
+    name: "IntSetObject",
+    properties: {
+      intSet: "int<>",
+    },
+  };
+  interface IntSetObject {
+    intSet: Realm.Set<Realm.Types.Int>;
+  }
+
+  const teamSchema = {
+    name: "Team",
+    properties: {
+      names: "string<>",
+    },
+  };
+  interface Team {
+    names: Realm.Set<string>;
+  }
+
+  describe("Schema", () => {
+    it("short (JS) and canonical schema types yield the same results", () => {
+      const shorthandSchema = IntSetObjectSchema;
+
+      const canonicalSchema = {
+        name: "CanonicalSchema",
+        properties: {
+          intSet: { type: "set", objectType: "int" },
+        },
+      };
+
+      const shorthandRealm = new Realm({ schema: [shorthandSchema] });
+      const shSchema = shorthandRealm.schema[0];
+      shorthandRealm.close();
+
+      const canonicalRealm = new Realm({ schema: [canonicalSchema] });
+      const canSchema = canonicalRealm.schema[0];
+      canonicalRealm.close();
+      expect(canSchema.properties).deep.equal(
+        shSchema.properties,
+        "Canonical and shorthand schemas should have idendical properties",
+      );
+    });
+  });
+
+  describe("General functionality", () => {
+    it("should add and delete operations", () => {
+      const realm = new Realm({ schema: [teamSchema] });
+
+      realm.write(() => {
+        // insert two people
+        realm.create(teamSchema.name, {
+          names: ["John", "Sue"],
+        });
+      });
+
+      let teams = realm.objects<Team>(teamSchema.name);
+      expect(teams.length).equals(1, "There should be one team");
+      expect(teams[0].names.size).equals(2, "Team Set size should be 2");
```

Comment:
This way we avoid implicitly testing `Results` too.

```suggestion
      const team = realm.write(() => {
        // insert two people
        return realm.create(teamSchema.name, {
          names: ["John", "Sue"],
        });
      });

      expect(team.names.size).equals(2, "Team Set size should be 2");
```